bpfd-api/src/bpfd.v1.rs

```diff
@@ -1,41 +1,63 @@
 #[derive(Clone, PartialEq, ::prost::Message)]
-pub struct LoadRequest {
+pub struct LoadRequestCommon {
     #[prost(string, tag = "1")]
     pub location: ::prost::alloc::string::String,
     #[prost(string, tag = "2")]
     pub section_name: ::prost::alloc::string::String,
-    #[prost(enumeration = "ProgramType", tag = "3")]
+    #[prost(int32, tag = "3")]
     pub program_type: i32,
-    #[prost(oneof = "load_request::AttachType", tags = "4, 5")]
-    pub attach_type: ::core::option::Option<load_request::AttachType>,
+    #[prost(string, optional, tag = "4")]
+    pub id: ::core::option::Option<::prost::alloc::string::String>,
 }
-/// Nested message and enum types in `LoadRequest`.
-pub mod load_request {
-    #[derive(Clone, PartialEq, ::prost::Oneof)]
-    pub enum AttachType {
-        #[prost(message, tag = "4")]
-        NetworkMultiAttach(super::NetworkMultiAttach),
-        #[prost(message, tag = "5")]
-        SingleAttach(super::SingleAttach),
-    }
+#[derive(Clone, PartialEq, ::prost::Message)]
+pub struct NoAttachInfo {}
+#[derive(Clone, PartialEq, ::prost::Message)]
+pub struct XdpAttachInfo {
+    #[prost(int32, tag = "1")]
+    pub priority: i32,
+    #[prost(string, tag = "2")]
+    pub iface: ::prost::alloc::string::String,
+    #[prost(int32, tag = "3")]
+    pub position: i32,
```

Comment:
`position` is read-only, correct (set by bpfd)? Are there any others? If so, should they be in a separate struct or is there some way to mark as read-only in the proto definition?

Reply:
There isn't a way to make this RO.
We *could* use a separate struct, but I decided it wasn't worth it - we'll just ignore it if it's set in a request.